Common/Classes/TaggedException.cs

```diff
@@ -151,6 +153,7 @@ public TaggedException(T tag, string message, System.Exception innerException)
         /// <param name="tag"></param>
         /// <param name="info"><see cref="System.Runtime.Serialization.SerializationInfo"/></param>
         /// <param name="context"><see cref="System.Runtime.Serialization.StreamingContext"/></param>
+        [Obsolete]
```

Comment:
👍 It served it purpose well, if its going to be obsoleted we should probably provide a small guidance comment on how to throw and exception and catch it with user supplied data or make such an example used in the library and refer to it in the comment so there is less confusion.